src/Grocery/Grocery.js

```diff
@@ -1,34 +1,131 @@
 import React, { Component } from 'react';
-import { CheckBox } from 'react-native-elements';
-
 import Meteor from 'react-native-meteor';
 
+import { colors, stylesheet } from '../../config/styles';
+
+import { TextInput, SafeAreaView, Alert, Text } from 'react-native';
+import { Button } from 'react-native-elements';
+
 export default class Grocery extends Component {
+	
 	constructor(props) {
 		super(props);
+		
+		const groceryId = props.navigation.state.params.id;
+		let name = '';
+		let amount = null;
+		let newGrocery = true;
+		if (groceryId) {
+			newGrocery = false;
+
+			const grocery = this.props.screenProps.groceries.find(grocery => grocery._id === groceryId);
+			if (grocery) {
+				name = grocery.name;
+				amount = grocery.amount;
+			}
+		}
+		this.state = {
+			listId: props.navigation.state.params.listId,
+			name,
+			amount,
+			newGrocery
+		}
 	}
+    
+	addGrocery() {
+		if (this.state.name.length === 0) {
+			Alert.alert(
+				'Error updating Grocery item',
+				'Grocery name must not be empty',
+				[
+					{ text: "OK", style: 'normal'}
+				],
+				{ cancelable: true }
+			);
+		}
 
-	handlePress() {
-		let checked = !this.props.item.checked;
-		Meteor.call('groceries.setChecked', this.props.item._id, checked);
-		
-		if ( checked ) {
-			Meteor.call('inventories.insert', this.props.item.name);
+		Meteor.call('groceries.insert', this.state.name, this.state.amount, (err, groceryId) => {
+			if (err) {
+				Alert.alert(
+					'Error creating Grocery item',
+					err.error,
+					[
+						{ text: "OK", style: 'normal' }
+					],
+					{ cancelable: true }
+				);
+				return;
+			}
+
+			Meteor.call('grocerylists.addItem', this.state.listId, groceryId);
+			this.props.navigation.goBack();
+		});
+	}
+
+	updateGrocery() {
```

Comment:
Updating amount doesn't appear to work here. I updated the amount and it rewinds back to the previous screen but the cell still has the old amount/name shown